src/wp-includes/class-wp-query.php

```diff
@@ -4838,6 +4838,7 @@ protected function generate_cache_key( array $args, $sql ) {
 		global $wpdb;
 
 		unset(
+			$args['post_type'],
```

Comment:
We could rather remove `$post_type` from key generation. The key generation does depend on sql, so have a different hash key generated for different $post_type.